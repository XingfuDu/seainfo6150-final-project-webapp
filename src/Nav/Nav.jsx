import React from 'react';
import { Link } from "react-router-dom";
import classes from "./Nav.module.css";

const Nav = () => {
    return (
        <nav className={classes.navBar}>
            <ul className={classes.ul}>
                <li className={classes.li}>
                    <Link className={classes.link} to="/">Home</Link>
                </li>
                <li className={classes.li}>
                    <Link className={classes.link} to="/category">Category</Link>
                </li>
                <li className={classes.li}>
                    <Link className={classes.link} to="/aboutUs">About Us</Link>
                </li>
                <li className={classes.li}>
                    <Link className={classes.link} to="/contact">Contact</Link>
                </li>
            </ul>
        </nav>
    );
};

export default Nav;