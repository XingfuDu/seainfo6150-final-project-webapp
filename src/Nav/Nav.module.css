.navBar {
    display: flex;
    justify-content: space-between;
    background-color: #293241;
    font-family: Arial, Helvetica, sans-serif;
    height: 100px;
    width: 100%;
    position: fixed;
    top: 0;
    z-index: 99;
}

.ul {
    display: flex;
    flex: 1;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
    padding-left: 0;
    margin-left: 20px;
    margin-right: 20px;
}

.ul .li {
    display: inline;
    margin: 0;
    padding: 0;
}

.link {
    color: white;
    padding: 15px;
    border-radius: 5px;
    background-color: #ce5e42;
    text-decoration: none;
    font-size: 150%;
}

.link:hover {
    background-color: #f8704e; 
}

@media (max-width: 1000px) {
    .navBar {
        height: 400px;
    }
    .ul {
        display:flex;
        flex-direction: column;
        align-items: flex-start;
        justify-content: space-around;
    }
}