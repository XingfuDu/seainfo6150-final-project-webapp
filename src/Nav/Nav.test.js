import React from 'react';
import Nav from './Nav';
import { render } from "@testing-library/react";
import { BrowserRouter } from 'react-router-dom';

describe('Nav tests', () => {
	it('renders correctly', () => {
		const { container } = render(<BrowserRouter><Nav/></BrowserRouter>);
		expect(container).toMatchSnapshot();
	});
});
