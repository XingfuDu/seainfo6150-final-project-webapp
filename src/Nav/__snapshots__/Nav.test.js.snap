// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav tests renders correctly 1`] = `
<div>
  <nav
    class="navBar"
  >
    <ul
      class="ul"
    >
      <li
        class="li"
      >
        <div
          class="wrapper"
        >
          <img
            alt=""
            class="logo"
            src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSEbIOIbjScifPvLGXY2gwa22xCLe780nRyYQ&usqp=CAUÃŸ"
          />
          <a
            class="link"
            href="/home"
          >
            Book Store
          </a>
        </div>
      </li>
      <li
        class="li"
      >
        <a
          class="link"
          href="/category"
        >
          Category
        </a>
      </li>
      <li
        class="li"
      >
        <a
          class="link"
          href="/aboutUs"
        >
          About Us
        </a>
      </li>
      <li
        class="li"
      >
        <a
          class="link"
          href="/contact"
        >
          Contact
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
